### PR TITLE
Jsonlog

### PR DESCRIPTION
**Pull request by mattretzer**
Hi @nelliemckesson, I ran this through dev_testing and 3 other project folders on staging; the output looks good :)

### CODE REVIEW

File: bookmaker_titlepage.rb [hunk 1]
```diff
@@ -3,81 +3,79 @@
 require 'fileutils'
 
 require_relative '../bookmaker/core/header.rb'
+require_relative '../bookmaker/core/metadata.rb'
```
Comment (nelliemckesson) [inline bookmaker_titlepage.rb]: FYI, currently in most deployments, bookmaker_titlepage runs _before_ metadata_preprocessing (which is what gathers most of the metadata).

File: bookmaker_titlepage.rb [hunk 2]
```diff
@@ -93,42 +91,223 @@ def getResourceDir(imprint, json)
...
+  return ''
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
 
-# --------------- ISBN FINDER COPIED FROM BOOKMAKER_ADDONS/METADATA_PREPROCESSING
-# testing to see if ISBN style exists
-spanisbn = File.read(Bkmkr::Paths.outputtmp_html).scan(/spanISBNisbn/)
-multiple_isbns = File.read(Bkmkr::Paths.outputtmp_html).scan(/spanISBNisbn">\s*.+<\/span>\s*\(((hardcover)|(trade\s*paperback)|(mass.market.paperback)|(print.on.demand)|(e\s*-*\s*book))\)/)
+def setFinalCover(oldtitlepage, newtitlepage, logkey='')
```
Comment (nelliemckesson) [inline bookmaker_titlepage.rb]: rename this function "setFinalTitlepage"

File: bookmaker_titlepage.rb [hunk 3]
```diff
@@ -93,42 +91,223 @@ def getResourceDir(imprint, json)
...
+  end
+  return final_cover
+rescue => logstring
+  return ''
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+# Determine whether or not to generate a titlepage.
+def decideTitlepageGeneration(titlepagelog, newtitlepage, oldtitlepage, final_cover, logkey='')
```
Comment (nelliemckesson) [inline bookmaker_titlepage.rb]: rename to "detectAutoGeneratedTitlepage" to match covermaker function name (or vice versa)

File: bookmaker_titlepage.rb [hunk 4]
```diff
@@ -93,42 +91,223 @@ def getResourceDir(imprint, json)
...
+    gen = true
+  end
+  return gen
+rescue => logstring
+  return ''
+ensure
+  Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+# def chooseCss(project_dir, pdf_css_dir, logkey='')
```
Comment (nelliemckesson) [inline bookmaker_titlepage.rb]: Is this commented out because it is not necessary? If so, delete.

File: bookmaker_titlepage.rb [hunk 5]
```diff
@@ -93,42 +91,223 @@ def getResourceDir(imprint, json)
...
+
+## wrapping a Mcmlln::Tools method in a new method for this script; to return a result for json_logfile
+def rmFile(file, logkey='')
+	Mcmlln::Tools.deleteFile(file)
+rescue => logstring
+ensure
+	Mcmlln::Tools.logtoJson(@log_hash, logkey, logstring)
+end
+
+def titlepageTest(final_cover, logkey='')
```
Comment (nelliemckesson) [inline bookmaker_titlepage.rb]: rename to "titlepageTests" to match covermaker method naming.

File: bookmaker_titlepage.rb [hunk 6]
```diff
@@ -153,152 +332,92 @@ def getResourceDir(imprint, json)
...
-  Mcmlln::Tools.deleteFile(titlepagelog)
-  Mcmlln::Tools.deleteFile(oldtitlepage)
-# Finally: if no titlepage has ever been generated, and no new image has been submitted,
-# and there is no existing image archived from a previous run
-elsif !File.file?(titlepagelog) and !File.file?(final_cover)
-  gen = true
-end
-
+gen = decideTitlepageGeneration(titlepagelog, newtitlepage, oldtitlepage, final_cover, 'determine_titlepage_generation')
+@log_hash['gen(erate_titlepage)_value'] = gen
```
Comment (nelliemckesson) [inline bookmaker_titlepage.rb]: what's up with these weird parentheses?

File: bookmaker_titlepage.rb [hunk 7]
```diff
@@ -153,152 +332,92 @@ def getResourceDir(imprint, json)
...
 end
 
 # CSS that will format the final titlepage PDF
-if File.file?("#{Bkmkr::Paths.scripts_dir}/covermaker/css/#{project_dir}/titlepage.css")
-  cover_css_file = File.join(pdf_css_dir, project_dir, "titlepage.css")
-else
-  cover_css_file = File.join(pdf_css_dir, "generic", "titlepage.css")
-end
+#this method was checking the css path based project_dir from metadata, which at this stage _
+#in our deploy.bad only holds 'N/A'.  So commenting out the chooseCSS method and it's call here:
```
Comment (nelliemckesson) [inline bookmaker_titlepage.rb]: deploy.bad > deploy.bat?

File: bookmaker_titlepage.rb [hunk 8]
```diff
@@ -153,152 +332,92 @@ def getResourceDir(imprint, json)
...
   final_cover = arch_epubtp
 end
 
 # CSS that will format the final titlepage PDF
-if File.file?("#{Bkmkr::Paths.scripts_dir}/covermaker/css/#{project_dir}/titlepage.css")
-  cover_css_file = File.join(pdf_css_dir, project_dir, "titlepage.css")
-else
-  cover_css_file = File.join(pdf_css_dir, "generic", "titlepage.css")
-end
+#this method was checking the css path based project_dir from metadata, which at this stage _
```
Comment (nelliemckesson) [inline bookmaker_titlepage.rb]: "based project_dir" > "based on the project_dir"?

File: bookmaker_titlepage.rb [hunk 9]
```diff
@@ -153,152 +332,92 @@ def getResourceDir(imprint, json)
...
+#in our deploy.bad only holds 'N/A'.  So commenting out the chooseCSS method and it's call here:
+# cover_css_file = chooseCss(project_dir, pdf_css_dir, 'choose_cover_css_file')
```
Comment (nelliemckesson) [inline bookmaker_titlepage.rb]: Also, this answers my question from above. So, as long as you test and everything seems to be working, you can delete the old method.

File: bookmaker_titlepage.rb (hunk 6 above)
Comment (mattretzer) [inline bookmaker_titlepage.rb]: I was explaining what 'gen' meant to aid in readability of jsonlogfile, but trying to still show it was the value of 'gen'.  How about "gen_value_(generate_titlepage_bool)"?

**Pull request merged**